website/docs/reference/rbac.md

```diff
@@ -19,22 +19,160 @@ Unleash has two levels in its hierarchy of resources:
 
+## Custom Root Roles
+
+:::info availability
+
+Custom root roles were introduced in **Unleash 5.4** and are only available in Unleash Enterprise.
+
+:::
+
+Custom root roles let you define your own root roles with a specific set of root permissions. The roles can then be assigned to entities (users, service accounts and groups) at the root level. This allows you to control access to resources in a more precise, fine-grained way. For a step-by-step walkthrough of how to create and assign custom root roles, see [_how to create and assign custom root roles_](../how-to/how-to-create-and-assign-custom-root-roles.md).
+
+Each custom root role consists of:
+
+- a **name** (required)
+- a **role description** (required)
+- a set of **root permissions** (required)
+
+### Root permissions
+
+You can assign the following root permissions:
+
+#### Addon permissions
+
+- **Create addons**
+
+  Lets the user create addons.
+
+- **Update addons**
+
+  Lets the user update addons.
+
+- **Delete addons**
+
+  Lets the user delete addons.
+
+#### API token permissions
+
+- **Read frontend API tokens**
+
+  Lets the user read frontend API tokens.
+
+- **Create frontend API tokens**
+
+  Lets the user create frontend API tokens.
+
+- **Update frontend API tokens**
+
+  Lets the user update frontend API tokens.
+
+- **Delete frontend API tokens**
+
+  Lets the user delete frontend API tokens.
+
+- **Read client API tokens**
+
+  Lets the user read client API tokens.
+
+- **Create client API tokens**
+
+  Lets the user create client API tokens.
+
+- **Update client API tokens**
+
+  Lets the user update client API tokens.
+
+- **Delete client API tokens**
+
+  Lets the user delete client API tokens.
+
+#### Application permissions
+
+- **Update applications**
+
+  Lets the user update applications.
+
+#### Context field permissions
+
+- **Create context fields**
+
+  Lets the user create context fields.
+
+- **Update context fields**
+    
+  Lets the user update context fields.
+
+- **Delete context fields**
+
+  Lets the user delete context fields.
+
+#### Project permissions
+
+- **Create projects**
+
+  Lets the user create projects.
+
+#### Role permissions
+
+- **Read roles**
+
+  Lets the user read roles.
+
+#### Segment permissions
+
+- **Create segments**
+    
+  Lets the user create segments.
+
+- **Edit segments**
+
+  Lets the user edit segments.
+
+- **Delete segments**
+    
+  Lets the user delete segments.
+
+#### Strategy permissions
+
+- **Create strategies**
+
+  Lets the user create strategies.
+
+- **Update strategies**
+
+  Lets the user update strategies.
+
+- **Delete strategies**
+
+  Lets the user delete strategies.
+
+#### Tag type permissions
+
+- **Update tag types**
+
+  Lets the user update tag types.
+
+- **Delete tag types**
+
+  Lets the user delete tag types.
+
```

Comment:
Non-blocking, but we might want to consider using [description lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl) or tables here. That would create stronger semantic links between the permission name and its definition. 

Unfortunately, markdown doesn't have a syntax for description lists, so we'd have to drop down to raw html for it. That makes editing it a little trickier, but I think it's a fair tradeoff.

What do you think?

Reply:
Yeah, that's very nice! 😄  At least it looks very nice. I'm a little worried about the semantics for people who can't see the table. How do screen readers treat empty columns? Probably not as 'repeat the last non-empty entry in this column'. So how about splitting it up into multiple tables, each under its own heading? (keeping the heading structure from before)

Reply:
Addressed in https://github.com/Unleash/unleash/pull/4451/commits/f9e601be7af170d3371d71259f6aff49162d1102